src/pages/en/guides/typescript.md

```diff
@@ -103,6 +103,12 @@ const { greeting = 'Hello', name } = Astro.props;
 <h2>{greeting}, {name}!</h2>
 ```
 
+:::tip
+
+To type a component that denies props, you can pass `type Props = Record<string, never>;` for zero props and `type Props = { children: any; };` for no props except for the default slot.
```

Comment:
It's a bit annoying that using `type Props = Record<string, never>;` means you have to redefine `children`. This is not the case when doing `type Props = {};`

In my opinion, the ESLint rule is wrong here (and in general, for a JSX definition) 🤷‍♀️

Reply:
I think it makes sense that you need to be specific here... in a function based JSX context like react function components, you just wouldn't accept a props argument to the function.

There's an argument for always requiring the user to specify the type of children rather than injecting it

react 18 removed implicit children: https://solverfox.dev/writing/no-implicit-children/

Reply:
Where currently astro works like `React.FC<Props>`, while I'd like it to function like `Props`, where no props or children are accepted unless specified in the file, hopefully with a nice astro type to import for the type of the children/slots

Reply:
though i see how this is an incremental thing, as we move people onto the idea 👍

Reply:
I agree with you that if we want to be very strict, it's much better to not have any implicit props. It's annoying in Astro context where we both want to support being very strict, but also not strict at all

Reply:
It'd be great if that was configurable somehow maybe?? i dunno 😔